### DISCUSSION TITLE
Option to keep 1 of each item in backpack for upgrades that pull out items

### DISCUSSION THREAD

**Discussion by MindCaged** [P3pp3rF1y/SophisticatedBackpacks#108] [Ideas]
For upgrades like Deposit, Filter, Refill, etc. It would be nice to have an option to prevent pulling out the last of any given item when you're using the match backpack contents option for your pickup/magnet upgrade.

**Comment by bondenn** [comment]
I agree with this suggestion, in addition, it would be nice if there was a setting for the compacting upgrade to only compact items when the stack is 10 or higher so that it won't compact all the items in the backpack causing your inventory to get clogged.

**Reply by natelovesyou** [reply]
> I agree with this suggestion, in addition, it would be nice if there was a setting for the compacting upgrade to only compact items when the stack is 10 or higher so that it won't compact all the items in the backpack causing your inventory to get clogged.

I agree with both ideas, but especially the compacting one. It's counter-intuitive to use up the last of an item when auto-compacting so that the pack fails to auto-pickup that item again.
